osu.Game.Rulesets.Osu/Difficulty/Evaluators/AimEvaluator.cs

```diff
@@ -112,18 +121,48 @@ public static double EvaluateDifficultyOf(DifficultyHitObject current, bool with
             }
 
-            if (osuLastObj.BaseObject is Slider)
+            // Add in acute angle bonus or wide angle bonus + velocity change bonus, whichever is larger.
+            aimStrain += Math.Max(acuteAngleBonus * acute_angle_multiplier, wideAngleBonus * wide_angle_multiplier + velocityChangeBonus * velocity_change_multiplier);
+
+            double sliderJumpBonus = 0;
+            if (osuLastObj.BaseObject is Slider slider)
             {
-                // Reward sliders based on velocity.
-                sliderBonus = osuLastObj.TravelDistance / osuLastObj.TravelTime;
+                // Take just slider heads into account because we're computing sliderjumps, not slideraim
+                sliderJumpBonus = slider_jump_multiplier * aimStrain * sliderJumpsAdjustingMultiplier;
+
+                // Reward more if sliders and circles are alternating (actually it's still lower than several sliders in a row)
+                if (osuLastLastObj.BaseObject is HitCircle)
+                {
+                    double alternatingBonus = slider_jump_multiplier * osuLastObj.JumpDistance / osuLastObj.StrainTime;
+
+                    if (osuLastObj.StrainTime > osuLastLastObj.StrainTime)
+                        alternatingBonus *= Math.Pow(Math.Min(osuCurrObj.StrainTime, osuLastObj.StrainTime) / Math.Max(osuCurrObj.StrainTime, osuLastObj.StrainTime), 2);
+
+                    sliderJumpBonus += alternatingBonus;
+                }
+
+                // If slider was slower than notes before - punish it
+                if (osuCurrObj.StrainTime > osuLastObj.StrainTime)
+                    sliderJumpBonus *= Math.Pow(Math.Min(osuCurrObj.StrainTime, osuLastObj.StrainTime) / Math.Max(osuCurrObj.StrainTime, osuLastObj.StrainTime), 2);
+
+                // Punish too short sliders to prevent cheesing (cheesing is still possible, but it's very rare)
+                double sliderLength = slider.Velocity * slider.SpanDuration;
+                if (sliderLength < slider.Radius)
+                    sliderJumpBonus *= sliderLength / slider.Radius;
             }
 
-            // Add in acute angle bonus or wide angle bonus + velocity change bonus, whichever is larger.
-            aimStrain += Math.Max(acuteAngleBonus * acute_angle_multiplier, wideAngleBonus * wide_angle_multiplier + velocityChangeBonus * velocity_change_multiplier);
+            aimStrain += sliderJumpBonus;
 
             // Add in additional slider velocity bonus.
-            if (withSliderTravelDistance)
-                aimStrain += sliderBonus * slider_multiplier;
+            double sliderBonus = 0;
+
+            if (withSliderTravelDistance && osuLastObj.BaseObject is Slider)
+            {
+                // Reward sliders based on velocity.
+                sliderBonus = osuLastObj.TravelDistance / osuLastObj.TravelTime;
+            }
+
+            aimStrain += sliderBonus * slider_aim_multiplier;
```

Comment:
if the slider jump bonus and slider aim bonus are not meant to be applied together, then i would suggest throwing them into a max function. this would communicate their purpose much better, and it might render `sliderJumpsAdjustingMultiplier` redundant. 

so i'd be expecting something like:
`aimStrain += Math.max(sliderBonus * slider_aim_multiplier, sliderJumpBonus);`

Reply:
they meant to be applied together
sliderjumps account for the fact that aiming sliderhead is harder than normal circle
sliderbody is aiming the body after you hit the circle
sliderJumpsAdjustingMultiplier exist because i don't want sliderbody difficulty to be multiplied by sliderjumps bonus

Reply:
so they are meant to be applied together, but also there is a nerf factor that "exists to prevent slideraim having sliderjumps bonus" and you "don't want sliderbody difficulty to be multiplied by sliderjumps bonus". this seems to be a pretty confusing contradiction on a conceptual basis, and i wouldn't expect anyone coming in with a fresh pair of eyes to be able to comprehend what's going on based on the code alone...

Reply:
in my opinion it's pretty simple to understand that sliderjump bonus is for aiming to the circle, while slideraim is for aiming the body
aiming the body doesn't gets any harder if previous object was a slider, so i can't just use raw aim strain that includes body difficulty

Reply:
i fully understand what slider jump bonus and slider aim bonus is, that doesn't really address my point of there seemingly being contradictions in approach here. if there are no contradictions here and i am mistaken, then the changes here need to be made much clearer